analytics/config.go

```diff
@@ -0,0 +1,96 @@
+package analytics
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/google/uuid"
+	"gopkg.in/yaml.v2"
+)
+
+// Config represents Wash's Google Analytics Config
+type Config struct {
+	Disabled bool      `yaml:"disabled"`
+	UserID   uuid.UUID `yaml:"user-id"`
+}
+
+// GetConfig returns Wash's analytics config, which is located at
+// $HOME/.puppetlabs/wash/analytics.yaml.
+func GetConfig() (config Config, err error) {
+	// Check if analytics is disabled via the environment before
+	// returning
```

Comment:
The defer enforces precedence -- disabling analytics via environment has greater precedence than everything else. Otherwise, if we check environment first, then unmarshalling could overwrite our disabled value and the additional logic to check that could be annoying.

Will add a comment about this if you think it is OK.

Reply:
Why not just return immediately if the environment variable is set? If it's set we don't need an ID and I don't see a reason to create the analytics file.

Reply:
That also makes sense. Will update.